libraries/chain/custom_authority_evaluator.cpp

```diff
@@ -87,29 +55,38 @@ object_id_type custom_authority_create_evaluator::do_apply(const custom_authorit
 
 void_result custom_authority_update_evaluator::do_evaluate(const custom_authority_update_operation& op)
 { try {
-   const database& d = db();
-
+   FC_ASSERT(db().head_block_time() > HARDFORK_CORE_1285_TIME, "custom_authority_update_operation should not be executed before HARDFORK_CORE_1285_TIME");
    return void_result();
 } FC_CAPTURE_AND_RETHROW( (op) ) }
 
 void_result custom_authority_update_evaluator::do_apply(const custom_authority_update_operation& op)
 { try {
    database& d = db();
-
+   
+   d.modify( d.get<custom_authority_object>(op.custom_authority_to_update), [&]( custom_authority_object& obj ){
```

Comment:
Yes db.get implicitly checks, however, throwing when calling db.modify makes the process a bit complicated, because undo_db stuff will be executed when calling modify thus may impact performance.

It's an architectural decision that is specific to our code base: in do_evaluate we check as many things as possible and prepare for update, in do_apply we try to only update the database. More discussion in https://github.com/bitshares/bitshares-core/issues/1318.

Reply:
Ah interesting, this is nice to know. Thx.

Reply:
done